File: practica1-java/src/gestorAplicacion/obras/Folleto.java
package gestorAplicacion.obras;

import java.util.ArrayList;

public class Folleto extends Publicacion {
	// ATRIBUTOS DE CLASE
	private static final long serialVersionUID = 1L;
	static private ArrayList<Folleto> folletos = new ArrayList<Folleto>();

	// ATRIBUTOS INSTANCIA
	private String referencia;

	// CONSTRUCTORES
	public Folleto(int codigo, String nombre, short año, short ejemplar, String referencia,Estanteria estanteria) {
		super(codigo, nombre, año, ejemplar);
		this.referencia = referencia;
		super.estanteria = estanteria;
		folletos.add(this);
		estanteria.getPublicaciones().add(this);
		Publicacion.numeroPublicaciones++;
	}

	// METODOS
	public String mostrarInfo() {
		return " INFORMACION DEL FOLLETO " + "\n" + "Nombre: " + this.nombre + "\n" + "Año: " + this.año + "\n" 
				+ "Codigo: " + this.codigo + "\n" + "Ejemplar: " + this.ejemplar + "\n" + "Estado: " + this.estado
				+ "\n" + "Referencia: " + this.referencia;
	}

	public String mostrarUbicacion() {
		return "Se localiza en -> " + this.estanteria.mostrarInfo();
	}

	// GETTERS Y SETTERS
	public String getReferencia() {
		return referencia;
	}

	public void setReferencia(String referencia) {
		this.referencia = referencia;
	}

	public static ArrayList<Folleto> getFolletos() {
		return folletos;
	}

	public static void setFolletos(ArrayList<Folleto> folletos) {
		Folleto.folletos = folletos;
	}

}

File: practica2-python/src/gestorAplicacion/obras/Folleto.py
from gestorAplicacion.obras.Publicacion import Publicacion,Estado

class Folleto(Publicacion): 
    
    # serialversion ?

    #Atributos de clase
    
    _folleto = []
    

    #Constructor
    
    def __init__(self, codigo,nombre,año,ejemplar,referencia='...',estanteria=None,prestamo=None) :
        Folleto._folleto.append(self)
        super().__init__(codigo,nombre,año,ejemplar)
        self._referencia = referencia
        self.estanteria = estanteria
        self.prestamo = prestamo
        
        
        if estanteria is not None:
            self.estanteria.getPublicaciones.append(self)

    #Metodos
    
    def mostrarInfo(self):
        return  f'INFORMACION DEL FOLLETO: \n Nombre: {self.getNombre()} \n Año: {self.getAño()} \n Codigo: {self.getCodigo()} \n Ejemplar: {self.getEjemplar()} \n Estado: {self.getEstado()} \n Referencia: {self.getReferencia()}'
        
    def mostrarUbicacion(self):
        return "Localizado en: " + str(self.estanteria.mostrarInfo())
    
    def getReferencia(self):
        return self._referencia
    def setReferencia(self,a):
        self._referencia = a

    @classmethod
    def getFolleto(cls):
        return Folleto._folleto
    @classmethod
    def setFolleto(cls,lista):
        Folleto._folleto = lista
